### PR TITLE
Bigtop build instructions for hbase

### PR DESCRIPTION
**Pull request by supreeth-t**
CL adds bigtop build instructions with minor changes.

@andy256 Please review.

cc @vjeyakum

### CODE REVIEW

File: bigtop-tests/test-artifacts/hbase/pom.xml [hunk 1]
```diff
@@ -72,5 +72,37 @@
...
+      <name>plugins-snapshot</name>
+      <url>http://artifactory.tetrationanalytics.com:8081/artifactory/plugins-snapshot</url>
+    </pluginRepository>
+  </pluginRepositories>
+
+  <distributionManagement>
+    <repository>
+      <id>central</id>
+      <name>artifactory.tetrationanalytics.com-releases</name>
+      <url>http://artifactory.tetrationanalytics.com:8081/artifactory/libs-release-local</url>
```
Comment (ghost) [inline bigtop-tests/test-artifacts/hbase/pom.xml]: make this ext-release-local

File: pom.xml [hunk 2]
```diff
@@ -521,8 +521,37 @@
...
+      <name>plugins-snapshot</name>
+      <url>http://artifactory.tetrationanalytics.com:8081/artifactory/plugins-snapshot</url>
+    </pluginRepository>
+  </pluginRepositories>
 
   <distributionManagement>
+    <repository>
+      <id>central</id>
+      <name>artifactory.tetrationanalytics.com-releases</name>
+      <url>http://artifactory.tetrationanalytics.com:8081/artifactory/libs-release-local</url>
```
Comment (ghost) [inline pom.xml]: and here

File: start-build-env.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ENV JAVA_HOME="/usr/lib/jvm/java-1.7.0-openjdk/"
+UserSpecificDocker
+
+# By mapping the .m2 directory you can do an mvn install from
+# within the container and use the result on your normal
+# system.  And this also is a significant speedup in subsequent
+# builds because the dependencies are downloaded only once.
+# -v "${HOME}/.m2:/home/${USER_NAME}/.m2" \
+# Add the above line if you want to use artifactory.tetration...
+# Does not work within VPN yet,, so kind of no-op for now.
```
Comment (ghost) [inline start-build-env.sh:67]: get on the docker for osx beta list! it works!

File: start-build-env.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+RUN yum -y install maven
+RUN yum -y install java-1.7.0-openjdk-devel.x86_64
+ENV JAVA_HOME="/usr/lib/jvm/java-1.7.0-openjdk/"
+UserSpecificDocker
+
+# By mapping the .m2 directory you can do an mvn install from
+# within the container and use the result on your normal
+# system.  And this also is a significant speedup in subsequent
+# builds because the dependencies are downloaded only once.
+# -v "${HOME}/.m2:/home/${USER_NAME}/.m2" \
```
Comment (ghost) [inline start-build-env.sh:65]: that's super neat

File: start-build-env.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+# Add the above line if you want to use artifactory.tetration...
+# Does not work within VPN yet,, so kind of no-op for now.
+docker run --rm=true -t -i \
+  -v "${PWD}/.:/home/${USER_NAME}/bigtop:z" \
```
Comment (ghost) [inline start-build-env.sh:69]: does this work, if the .m2 thing didn't work?

File: start-build-env.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+docker build -t "bigtop-build-${USER_NAME}" - <<UserSpecificDocker
+FROM centos
+RUN groupadd --non-unique -g ${GROUP_ID} ${USER_NAME}
+RUN useradd -g ${GROUP_ID} -u ${USER_ID} -k /root -m ${USER_NAME}
+ENV HOME /home/${USER_NAME}
+RUN yum -y install unzip
+RUN yum -y install rpm-build
+RUN yum -y install maven
+RUN yum -y install java-1.7.0-openjdk-devel.x86_64
```
Comment (ghost) [inline start-build-env.sh:57]: this could all be one yum install, slightly faster. /shrug

**Comment by ghost** [comment]
cool. LGTM modulo ext-release-local instead of libs-release-local -- this is technically an external project, so trying to keep artifactory separated that way.

**Comment by ghost** [comment]
just updated the intranet doc to be clear on that

File: start-build-env.sh (hunk 5 above)
Comment (supreeth-t) [inline start-build-env.sh:69]: .m2 thing will need vpn access. So commented out.

File: start-build-env.sh (hunk 3 above)
Comment (supreeth-t) [inline start-build-env.sh:67]: Will do

File: bigtop-tests/test-artifacts/hbase/pom.xml (hunk 1 above)
Comment (supreeth-t) [inline bigtop-tests/test-artifacts/hbase/pom.xml]: done

**Comment by ghost** [comment]
LGTM

**Pull request merged**